### ISSUE TITLE
MOxUnit seems to mask internal octave errors by using and empty catch statement

### ISSUE THREAD

**Issue by abravalheri** [MOxUnit/MOxUnit#39]
Please consider the code under [abravalheri/dummy-moxunit-example-for-issue](https://github.com/abravalheri/dummy-moxunit-example-for-issue/tree/issue-39) (branch `issue-39`), in order to reproduce this issue.

Despite of creating a test file that corresponds to the MOxUnit [conventions](https://github.com/MOxUnit/MOxUnit#defining-moxunit-tests), the output of `moxunit_runtests` is:

```bash
$ octave-cli --eval moxunit_runtests
suite: 0 tests

--------------------------------------------------

OK
ans = 1
```

This output  seems to indicate that MOxUnit does not recognize the existence of a test suite, which is very strange...

A deeper investigation shows that, MOxUnit does find the test suite, but an error occurs when running `initTestSuite`. One can hack the `MOxUnit/MOxUnit/initTestSuite.m` file and insert the following lines of code inside the empty `catch` statement (for the commit `3cd167f`, it corresponds to lines 81, 82):

```matlab
fprintf('----- Error! -----\n');
disp(lasterr);
fprintf('------------------\n\n');
```

Running the test again results in the following output:

```bash
$ octave-cli --eval moxunit_runtests
----- Error! -----
handles to nested functions are not yet supported
------------------

suite: 0 tests

--------------------------------------------------

OK
ans = 1
```

In summary, the generic `try ... catch ...` block, with and empty rescue section, masks the octave internal error, and makes debugging a nightmare.

Is there an specific reason for this block?

Maybe just removing it (and other generic `try. .. catch ...` that may exists) solves the problem.

**Comment by nno** [comment]
Thanks for the detailed report. Currently ``moxunit_run_tests`` tries to execute every .m file as a function in the current directory. The reason for a generic try-catch block was that there might be other, non-test, functions in that directory that could generate errors. I agree that this has the annoying side-effect that it masks errors for test functions.

On reconsideration, following what is done by nose (and possibly other test frameworks) it may be better to require that tests are stored in a file with name starting with 'test_' (or possibly, also allow ending with '_test'). Then, as you suggested, the try-catch block can be removed. What are your ideas about this approach?

**Comment by abravalheri** [comment]
@nno thank you as always for your kind response.

I think your idea about name conventions are the best way to go, since it avoid running unnecessary/unplanned  code. May I suggest just to drop the underscore character and make the comparison case insensitive (this way eventual class-style tests -- I really don't know if they are supported or will be -- or camelCased tests will be also recognized...).

Another consideration: I have tried to just comment the try/catch in the file `initTestSuite.m`, but it was not enough to make the error loud... I have also to comment another try/catch in the file `@MOxUnitTestNode/addFromFile.m`.

Maybe, despite of establishing a name convention, it would be useful to keep the try/catch statements... This way, even if some tests throw unexpected errors (e.g. wrong syntax, or octave lack of implementation), the others that are correct can still run. I think the point is reporting the error properly, instead of ignoring it...

**Comment by nno** [comment]
@abravalheri thanks for your helpful comments. I think it's a good idea to relax the naming convention a little - for example case-insensitive, starting with 'test', and the 5-th character either upper case or underscore.

I'll have a look at the try-catch statements and see what would be a good approach to report errors better.

**Comment by abravalheri** [comment]
Hi @nno,

I was working on a pull request about it. I will submit the PR for your consideration.
Please feel free to modify it and adequate to the conventions you prefer.

**Comment by nno** [comment]
This should be addressed by #42, please re-open if not.
